client/me/account/main.jsx

```diff
@@ -84,25 +82,58 @@ const Account = React.createClass( {
 		debug( this.constructor.displayName + ' component is unmounting.' );
 	},
 
-	updateLanguage() {
-		let valueLink = this.valueLink( 'language' );
+	getUserSetting( settingName ) {
+		return this.props.userSettings.getSetting( settingName );
+	},
 
-		valueLink.requestChange = ( value ) => {
-			const originalLanguage = this.props.userSettings.getOriginalSetting( 'language' );
+	updateUserSetting( settingName, value ) {
+		this.props.userSettings.updateSetting( settingName, value );
+	},
 
-			this.props.userSettings.updateSetting( 'language', value );
-			if ( value !== originalLanguage ) {
-				this.setState( { redirect: '/me/account' } );
-			} else {
-				this.setState( { redirect: false } );
-			}
-		};
+	updateUserSettingInput( settingName ) {
+		return event => this.updateUserSetting( settingName, event.target.value );
+	},
 
-		return valueLink;
+	updateUserSettingCheckbox( settingName ) {
+		return event => this.updateUserSetting( settingName, event.target.checked );
+	},
+
+	updateLanguage( event ) {
+		const { value } = event.target;
+		const originalLanguage = this.props.userSettings.getOriginalSetting( 'language' );
+
+		this.updateUserSetting( 'language', value );
+		if ( value !== originalLanguage ) {
+			this.setState( { redirect: '/me/account' } );
+		} else {
+			this.setState( { redirect: false } );
+		}
+	},
+
+	getEmailAddress() {
+		return this.hasPendingEmailChange()
+			? this.getUserSetting( 'new_user_email' )
+			: this.getUserSetting( 'user_email' );
+	},
+
+	updateEmailAddress( event ) {
+		const { value } = event.target;
+		if ( '' === value ) {
+			this.setState( { emailValidationError: 'empty' } );
+		} else if ( ! emailValidator.validate( value ) ) {
+			this.setState( { emailValidationError: 'invalid' } );
+		} else {
+			this.setState( { emailValidationError: false } );
+		}
```

Comment:
I'm not sure if this is really an improvement. The if/else chain might be verbose, but it's instantly clear what it's doing.

Maybe it would help if the branches were just setting a variable and there was one call to `this.setState` at the end?

Of course, this would be a great use case for a pattern-matching `switch` statement (like `match` in Rust) 😄